test/unit/mem/view/src/ViewStaticAccMem.cpp

```diff
@@ -16,92 +16,101 @@ using Elem = std::uint32_t;
 using Dim = alpaka::DimInt<2u>;
 using Idx = std::uint32_t;
 
-#if !defined(ALPAKA_ACC_SYCL_ENABLED)
+ALPAKA_STATIC_ACC_MEM_GLOBAL alpaka::DevGlobal<TAcc, Elem[3][2]> g_globalMemory2DUninitialized;
```

Comment:
@AuroraPerego We need to review this change again because of the bug shown https://github.com/alpaka-group/alpaka/issues/2259

In the original code we had constant global memory only and in the new code we have global memory and constant memory.
With you changes is there a difference of `ALPAKA_STATIC_ACC_MEM_GLOBAL` and `ALPAKA_STATIC_ACC_MEM_CONSTANT`?

Reply:
> In the original code we had constant global memory only and in the new code we have global memory and constant memory.

They were different also in the original code, with `ALPAKA_STATIC_ACC_MEM_GLOBAL` expanding to `__device__` and `ALPAKA_STATIC_ACC_MEM_CONSTANT` expanding to `__constant__`.

> With your changes is there a difference between ALPAKA_STATIC_ACC_MEM_GLOBAL and ALPAKA_STATIC_ACC_MEM_CONSTANT?

Yes, they are using the `__device__` and `__constant__` attributes respectively (plus the `inline` attribute)

> IMO in this section we should only use `ALPAKA_STATIC_ACC_MEM_CONSTANT` and later we have a test for `ALPAKA_STATIC_ACC_MEM_GLOBAL`

ok, but why? we were testing both here also before this PR

Reply:
> ok, but why? we were testing both here also before this PR

This is a good question.

What if we define `ALPAKA_STATIC_ACC_MEM_GLOBAL` static instead of `inline` this should create a single instance too.

Reply:
it compiles, I can make a PR with the change if you think it's correct (I don't know enough about `static`, `inline` and `extern` to be sure of that)

Reply:
In general, what macros should be used to support all use cases ?

### `file1.cu`
```c++
static __device__ int i = 0;
```

and

### `file2.h`
```c++
extern __device__ int i;
```

### `file2.cu`
```c++
#include "file2.h"

__device__ int i = 0;
```

?

Reply:
Thinking about it, maybe `static` is not what we want. The ideal would be to make the `extern` keyword work also with the new implementation to allow the `file2.h`/`file2.cu` case, which IIUC is not possible with `static`.